### DISCUSSION TITLE
Playlist Folder Support for Music Providers

### DISCUSSION THREAD

**Discussion by MizterB** [music-assistant/support#3039] [Feature requests and ideas]
_Submitting per recommendation by @MarvinSchenkel [on Discord](https://discord.com/channels/753947050995089438/1242741503642570752/1295994774918266881)_

Some music providers allow playlists to be organized into playlist folders, which greatly improves organization.  It would be useful to have the same concept supported in Music Assistant.  

Currently, Music Assistant 'flattens' all of a provider's playlists into a single level both within the Playlist and Browse views.  I would like to propose that at least the Browse view provide support for navigating a provider's playlist folder hierarchy if it has one.

In my current system, my Apple Music subscription contains over a thousand playlists, many with the same names.  By organizing them into folders, it is easier to impose my own navigation system to quickly find what I am looking for.  It has also eliminated the confusion of having many playlists with the same name at the same level of the hierarchy, and not knowing which playlist is which.

For Apple Music, it should be noted that Playlist Folders are not just an organizing construct, but are also considered a Playlists themselves.  Therefore, it is possible to queue a Playlist Folder, and the resulting queue will contain all child Playlists below it in the folder hierarchy.  I am unsure if this same concept exists for other providers, but I would also like to see this behavior extended into Music Assistant if the provider supports it.

There are some related feature requests / issues for specific providers:
#2894 - Allow Spotify Integration to view nested playlists.
#2472 - Concatenable/Nestable playlists

**Comment by soulstyle** [comment]
Really like to see this feature as well.
It would be great if this also included the possibility to create and organize your playlists in folders directly in MA :)

**Comment by Swimmer71** [comment]
This would be great. I just posted this comment here:

[https://github.com/orgs/music-assistant/discussions/1256#discussioncomment-11302278](url)

**Comment by cokefridge** [comment]
Absolutely. Folders are so underrated. Main reason for me to still use lyrion media server.

**Comment by arathilion** [comment]
I would also be very happy if the playlists could be displayed like in my app!

**Comment by dome2048** [comment]
Folders would be great and are really necessary for anyone with more than a handful of playlists.

**Reply by devedsmith** [reply]
I love this idea!

**Comment by HuggeK** [comment]
This would be great!

**Comment by Kludgy4** [comment]
I would like audiobooks, podcasts, and radio stations (as well as playlists) to be organisable by folder as well. Ideally I could mix-and-match these inside folders.

**Comment by lazyboy0172** [comment]
google brought me here when trying to find this functionality. Love MA, don't love scrolling through 100+ playlists in one view since now all of our music sources are consolidated to 1 app. Would love to be able to either import the folders from their sources or at least be able to create my own folder structure in MA to organize playlists

**Reply by OzGav** [reply]
In the meantime note that you can search.

**Reply by jacksaturn** [reply]
You say, "in the meantime..." Does that mean that this functionality is in the queue to be built?

I'm a brand new MA user — just set it up in the past few days, migrating from Audirvana.

Audirvana allows nested folders, and I've got 15+ years worth of individual monthly playlists (like "2026-09") which live inside yearly folders (like "2026"). All the year folders live inside one all-encompassing folder.

The benefit is threefold:
1) I don't have to navigate a vertical scroll of literally hundreds of playlists.
2) When I put on shuffle for a particular year's folder, the shuffled playlist includes all the tracks from across all of that year's monthly playlists.
3) When I put on shuffle for the master folder of all the years, I get a shuffle that contains *everything*.

Music Assistant has so much functionality that Audirvana and others lack, and I see more potential for these kind of positive updates happening with MA, which is encouraging. When MA introduces nested folders, I'm all in.

**Reply by OzGav** [reply]
No it isn’t the queue to be built but it is getting a few votes.
